plugins/updater/guest-js/index.ts

```diff
@@ -55,6 +56,27 @@ class Update extends Resource {
     this.body = metadata.body;
   }
 
+  /** Download the updater package */
+  async download(onEvent?: (progress: DownloadEvent) => void): Promise<void> {
+    const channel = new Channel<DownloadEvent>();
+    if (onEvent) {
+      channel.onmessage = onEvent;
+    }
+    this.downloadedBytesRid = await invoke<number>("plugin:updater|download", {
+      onEvent: channel,
+      rid: this.rid,
+    });
+  }
+
+  /** Install downloaded updater package */
+  async install(): Promise<void> {
+    await invoke("plugin:updater|install", {
```

Comment:
we should check if `this.downloadedBytesRid` is `undefined` and throw an error that `install` is used before `download`